### PR TITLE
Support embedded line breaks in headers, labels, and notes

### PR DESCRIPTION
**Pull request by dsliwka**
## Summary

- Adds support for embedded line breaks (`\n`) in row labels, column headers/spanners, row-group names, and notes, across the LaTeX, Typst, and Great Tables (HTML) output formats. DOCX already handled this natively via python-docx and needed no changes.
- Captions deliberately do **not** support line breaks - a caption is a single-line label by design; this was tried and reverted.
- A header vertical-centering feature (for rows mixing single-line and multi-line headers) was attempted and fully reverted: `\raisebox` silently does nothing inside LaTeX's tabularx X-columns (the default here), and the workaround needed to make it work there introduced its own bug (misaligned `\cmidrule`s). Not worth the added LaTeX fragility for a cosmetic detail - mismatched header rows sit flush at the top, as before.
- One known, accepted limitation: in HTML/Great Tables output, when one column header has a line break and a sibling doesn't, the spanner row below renders at slightly different vertical positions. This is Great Tables' own default rendering behavior; its public styling API (`tab_style` + `loc.spanner_labels()`) does not actually apply to this, so there is no clean fix available - not addressed here.
- Several robustness issues surfaced during review and manual PDF verification, all fixed with regression tests:
  - `notes=None` in Great Tables/HTML output rendered the literal text "None" instead of staying empty.
  - Typst silently dropped blank lines within multi-line text (e.g. `"Line one\n\nLine two"`), unlike tex and gt which both preserve the blank line.
  - A row-group name containing a line break crashed LaTeX compilation outright (`! Missing } inserted`) when the row-group format included a wrapping command like `\emph{...}` - caught only by actually compiling a descriptive-stats table with row groups, not by string-level testing.
  - A row-group name with a line break defaulted to makecell's own center-on-each-other line alignment instead of the left alignment used everywhere else (row labels, single-line group headers).

## Test plan

- [x] Full test suite passes (`pixi run -e dev tests`), including new regression tests in `tests/api/test_linebreaks.py` covering each position/format combination and the edge cases above.
- [x] Verified visually by rendering real tables (an `ETable` with line breaks in labels/headers/notes, and a `DTable` with row groups *and* column spanners whose names contain line breaks) to PDF across tex, typst, docx, and html, and inspecting the output.
- [x] Confirmed the row-group LaTeX fix actually compiles with `pdflatex` (not just checked at the string level), since that's exactly the gap that let the earlier crash slip through.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/maketables/mtable.py [hunk 1]
```diff
@@ -1134,15 +1136,25 @@ def _normalize_width(w: str | None) -> str | None:
...
+        # multicolumn, or row label) without it.
+        def _wrap_linebreaks(text: str, align: str | None = None) -> str:
+            text = text.replace("\n", r"\\")
+            if r"\\" in text:
+                # Omit the [align] spec for makecell's own default (center) so
+                # data-cell output (the common case) is byte-identical to
+                # before this function gained an align parameter.
+                spec = f"[{align}]" if align else ""
+                return f"\\makecell{spec}{{{text}}}"
+            return text
```
Comment (felixschmitz) [inline src/maketables/mtable.py]: Prefer a docstring over the leading multi-line comment for the function's purpose.

```suggestion
        def _wrap_linebreaks(text: str, align: str | None = None) -> str:
            """Convert embedded line breaks into LaTeX line breaks, wrapping in \\makecell{} when needed."""
            text = text.replace("\n", r"\\")
            if r"\\" in text:
                # Omit the [align] spec for makecell's own default (center) so
                # data-cell output (the common case) is byte-identical to
                # before this function gained an align parameter.
                spec = f"[{align}]" if align else ""
                return f"\\makecell{spec}{{{text}}}"
            return text
```

Comment (felixschmitz) [inline src/maketables/mtable.py]: Prefer a docstring over the leading multi-line comment for the function's purpose, and fold the `align` explanation into its Parameters section so the inline comment above `spec = ...` becomes obsolete too.

```suggestion
        def _wrap_linebreaks(text: str, align: str | None = None) -> str:
            """Convert embedded line breaks into LaTeX line breaks.

            Wraps in ``\\makecell{}`` when needed, since a bare ``\\\\``
            does not behave as a line break directly inside a tabular cell.

            Parameters
            ----------
            text : str
                Cell text, possibly containing newlines.
            align : str, optional
                Makecell alignment spec (e.g. ``"l"``). Omitted by default so
                the common case matches makecell's own default (center)
                output byte-for-byte.
            """
            text = text.replace("\n", r"\\")
            if r"\\" in text:
                spec = f"[{align}]" if align else ""
                return f"\\makecell{spec}{{{text}}}"
            return text
```

Comment (felixschmitz) [inline src/maketables/mtable.py]: Prefer a docstring over the leading multi-line comment for the function's purpose, and fold the `align` explanation into its Parameters section so the inline comment above `spec = ...` becomes obsolete too.

```suggestion
        def _wrap_linebreaks(text: str, align: str | None = None) -> str:
            """Convert embedded line breaks into LaTeX line breaks.

            Parameters
            ----------
            text : str
                Cell text, possibly containing newlines.
            align : str, optional
                Makecell alignment spec (e.g. ``"l"``). Omitted by default so
                the common case matches makecell's own default (center)
                output byte-for-byte.
            """
            text = text.replace("\n", r"\\")
            if r"\\" in text:
                spec = f"[{align}]" if align else ""
                return f"\\makecell{spec}{{{text}}}"
            return text
```

File: src/maketables/mtable.py [hunk 2]
```diff
@@ -1467,6 +1490,19 @@ def _prep_cell(x):
...
+        # row labels, headers/spanners, row-group names, caption, and notes.
+        def _join_typst_lines(content: str, escape_asterisks: bool = True) -> str:
+            if "\n" in content:
+                lines_content = content.split("\n")
+                escaped_lines = [
+                    self._escape_typst(line.strip(), escape_asterisks=escape_asterisks)
+                    for line in lines_content
+                ]
+                return " \\ ".join(escaped_lines)
+            return self._escape_typst(content, escape_asterisks=escape_asterisks)
```
Comment (felixschmitz) [inline src/maketables/mtable.py]: Same pattern here: move the leading comment into a docstring.

```suggestion
        def _join_typst_lines(content: str, escape_asterisks: bool = True) -> str:
            """Escape content and join embedded line breaks with Typst's line-break syntax.

            Parameters
            ----------
            content : str
                Text for any position (data cell, row label, header/spanner,
                row-group name, caption, or note), possibly containing newlines.
            escape_asterisks : bool, default True
                Whether to escape literal asterisks alongside other Typst
                special characters.
            """
            if "\n" in content:
                lines_content = content.split("\n")
                escaped_lines = [
                    self._escape_typst(line.strip(), escape_asterisks=escape_asterisks)
                    for line in lines_content
                ]
                return " \\ ".join(escaped_lines)
            return self._escape_typst(content, escape_asterisks=escape_asterisks)
```

File: src/maketables/mtable.py [hunk 3]
```diff
@@ -1666,6 +1690,30 @@ def _output_gt(
         # In all cells replace line breaks with <br>
         dfs = dfs.replace(r"\n", "<br>", regex=True)
 
+        # DataFrame.replace() above does not touch the index, so row labels
+        # (and row-group names) need the same conversion applied separately.
+        def _br(x):
+            return x.replace("\n", "<br>") if isinstance(x, str) else x
```
Comment (felixschmitz) [inline src/maketables/mtable.py]: Same here.

```suggestion
        def _br(x):
            """Convert embedded newlines in index values to ``<br>``.

            Notes
            -----
            ``DataFrame.replace()`` above only touches cell values, not the
            index, so row labels and row-group names need this separately.
            """
            return x.replace("\n", "<br>") if isinstance(x, str) else x
```

File: src/maketables/mtable.py [hunk 4]
```diff
@@ -1666,6 +1690,30 @@ def _output_gt(
...
+        # HTML-escape plain strings by default (unlike data cell values), so a
+        # literal "<br>" inserted there would show up as escaped, visible text
+        # rather than a real line break. Escape everything ourselves first,
+        # then wrap in great_tables.html() so GT trusts it and doesn't
+        # re-escape our deliberately-inserted <br> tags.
+        def _gt_linebreak(text):
+            if text is None:
+                return None
+            escaped = "<br>".join(_stdlib_html.escape(line) for line in str(text).split("\n"))
+            return _gt_html(escaped)
```
Comment (felixschmitz) [inline src/maketables/mtable.py]: And here.

```suggestion
        def _gt_linebreak(text):
            """Convert embedded newlines to a GT-trusted, ``<br>``-joined HTML string.

            Notes
            -----
            title/subtitle, cols_label, tab_spanner, and tab_source_note all
            HTML-escape plain strings by default (unlike data cell values), so
            a literal ``<br>`` would show up as visible, escaped text rather
            than a line break. Escape everything ourselves first, then wrap in
            ``great_tables.html()`` so GT trusts it and doesn't re-escape the
            deliberately-inserted ``<br>`` tags.
            """
            if text is None:
                return None
            escaped = "<br>".join(_stdlib_html.escape(line) for line in str(text).split("\n"))
            return _gt_html(escaped)
```

File: src/maketables/mtable.py (hunk 1 above)
Comment (felixschmitz) [inline src/maketables/mtable.py]: Prefer a docstring over the leading multi-line comment for the function's purpose, and fold the `align` explanation into its Parameters section so the inline comment above `spec = ...` becomes obsolete too.

```suggestion
        def _wrap_linebreaks(text: str, align: str | None = None) -> str:
            """Convert embedded line breaks into LaTeX line breaks.

            Wraps in ``\\makecell{}`` when needed, since a bare ``\\\\``
            does not behave as a line break directly inside a tabular cell.

            Parameters
            ----------
            text : str
                Cell text, possibly containing newlines.
            align : str, optional
                Makecell alignment spec (e.g. ``"l"``). Omitted by default so
                the common case matches makecell's own default (center)
                output byte-for-byte.
            """
            text = text.replace("\n", r"\\")
            if r"\\" in text:
                spec = f"[{align}]" if align else ""
                return f"\\makecell{spec}{{{text}}}"
            return text
```

File: src/maketables/mtable.py (hunk 2 above)
Comment (felixschmitz) [inline src/maketables/mtable.py]: Same pattern here: move the leading comment into a docstring.

```suggestion
        def _join_typst_lines(content: str, escape_asterisks: bool = True) -> str:
            """Escape content and join embedded line breaks with Typst's line-break syntax.

            Parameters
            ----------
            content : str
                Text for any position (data cell, row label, header/spanner,
                row-group name, caption, or note), possibly containing newlines.
            escape_asterisks : bool, default True
                Whether to escape literal asterisks alongside other Typst
                special characters.
            """
            if "\n" in content:
                lines_content = content.split("\n")
                escaped_lines = [
                    self._escape_typst(line.strip(), escape_asterisks=escape_asterisks)
                    for line in lines_content
                ]
                return " \\ ".join(escaped_lines)
            return self._escape_typst(content, escape_asterisks=escape_asterisks)
```

File: src/maketables/mtable.py (hunk 3 above)
Comment (felixschmitz) [inline src/maketables/mtable.py]: Same here.

```suggestion
        def _br(x):
            """Convert embedded newlines in index values to ``<br>``.

            Notes
            -----
            ``DataFrame.replace()`` above only touches cell values, not the
            index, so row labels and row-group names need this separately.
            """
            return x.replace("\n", "<br>") if isinstance(x, str) else x
```

File: src/maketables/mtable.py (hunk 4 above)
Comment (felixschmitz) [inline src/maketables/mtable.py]: Same here, and worth correcting the claim while we're at it: this comment says "title/subtitle" are covered, but `self.caption` is never actually passed through `_gt_linebreak()` (see `gt.tab_header(title=self.caption)` a few lines down, and the new test module's own docstring: "Captions do not support line breaks - a caption is a single-line label by design"). That's intentional per 447e917, so let's not imply otherwise.

```suggestion
        def _gt_linebreak(text):
            """Convert embedded newlines to a GT-trusted, ``<br>``-joined HTML string.

            Notes
            -----
            cols_label, tab_spanner, and tab_source_note all HTML-escape
            plain strings by default (unlike data cell values), so a literal
            ``<br>`` would show up as visible, escaped text rather than a
            line break. Escape everything ourselves first, then wrap in
            ``great_tables.html()`` so GT trusts it and doesn't re-escape
            the deliberately-inserted ``<br>`` tags. Not applied to the
            caption/title: captions are single-line labels by design.
            """
            if text is None:
                return None
            escaped = "<br>".join(_stdlib_html.escape(line) for line in str(text).split("\n"))
            return _gt_html(escaped)
```

File: src/maketables/mtable.py (hunk 1 above)
Comment (felixschmitz) [inline src/maketables/mtable.py:1151]: Prefer a docstring over the leading multi-line comment for the function's purpose, and fold the `align` explanation into its Parameters section so the inline comment above `spec = ...` becomes obsolete too.

```suggestion
        def _wrap_linebreaks(text: str, align: str | None = None) -> str:
            """Convert embedded line breaks into LaTeX line breaks.

            Wraps in ``\\makecell{}`` when needed, since a bare ``\\\\``
            does not behave as a line break directly inside a tabular cell.

            Parameters
            ----------
            text : str
                Cell text, possibly containing newlines.
            align : str, optional
                Makecell alignment spec (e.g. ``"l"``). Omitted by default so
                the common case matches makecell's own default (center)
                output byte-for-byte.
            """
            text = text.replace("\n", r"\\")
            if r"\\" in text:
                spec = f"[{align}]" if align else ""
                return f"\\makecell{spec}{{{text}}}"
            return text
```

File: src/maketables/mtable.py (hunk 2 above)
Comment (felixschmitz) [inline src/maketables/mtable.py:1504]: Same pattern here: move the leading comment into a docstring.

```suggestion
        def _join_typst_lines(content: str, escape_asterisks: bool = True) -> str:
            """Escape content and join embedded line breaks with Typst's line-break syntax.

            Parameters
            ----------
            content : str
                Text for any position (data cell, row label, header/spanner,
                row-group name, caption, or note), possibly containing newlines.
            escape_asterisks : bool, default True
                Whether to escape literal asterisks alongside other Typst
                special characters.
            """
            if "\n" in content:
                lines_content = content.split("\n")
                escaped_lines = [
                    self._escape_typst(line.strip(), escape_asterisks=escape_asterisks)
                    for line in lines_content
                ]
                return " \\ ".join(escaped_lines)
            return self._escape_typst(content, escape_asterisks=escape_asterisks)
```

File: src/maketables/mtable.py (hunk 3 above)
Comment (felixschmitz) [inline src/maketables/mtable.py:1696]: Same here.

```suggestion
        def _br(x):
            """Convert embedded newlines in index values to ``<br>``.

            Notes
            -----
            ``DataFrame.replace()`` above only touches cell values, not the
            index, so row labels and row-group names need this separately.
            """
            return x.replace("\n", "<br>") if isinstance(x, str) else x
```

File: src/maketables/mtable.py (hunk 4 above)
Comment (felixschmitz) [inline src/maketables/mtable.py:1715]: Same here, and worth correcting the claim while we're at it: this comment says "title/subtitle" are covered, but `self.caption` is never actually passed through `_gt_linebreak()` (see `gt.tab_header(title=self.caption)` a few lines down, and the new test module's own docstring: "Captions do not support line breaks - a caption is a single-line label by design"). That's intentional per 447e917, so let's not imply otherwise.

```suggestion
        def _gt_linebreak(text):
            """Convert embedded newlines to a GT-trusted, ``<br>``-joined HTML string.

            Notes
            -----
            cols_label, tab_spanner, and tab_source_note all HTML-escape
            plain strings by default (unlike data cell values), so a literal
            ``<br>`` would show up as visible, escaped text rather than a
            line break. Escape everything ourselves first, then wrap in
            ``great_tables.html()`` so GT trusts it and doesn't re-escape
            the deliberately-inserted ``<br>`` tags. Not applied to the
            caption/title: captions are single-line labels by design.
            """
            if text is None:
                return None
            escaped = "<br>".join(_stdlib_html.escape(line) for line in str(text).split("\n"))
            return _gt_html(escaped)
```

**Review by felixschmitz — APPROVED** [review]
